modules/Core/pages/user/messaging.php

```diff
@@ -406,6 +406,13 @@
                     // Display success message
                     $smarty->assign('MESSAGE_SENT', $language->get('user', 'message_sent_successfully'));
                     unset($_POST['content']);
+                    
+                    // Use strict comparison operator to compare request method
+                    if ($_SERVER['REQUEST_METHOD'] === 'POST') {
```

Comment:
There should be no need to do this check here as we'll already be within a POST request.